# The Correlation between Malaria and Insecticide treated mosquito nets in Kenya.
**Research Question**

Is there a correlation between health spending on Malaria, usage of insecticide treated bednets and reported presentation of malaria symtomps cases in the different counties in Kenya.

Question for consideration:


*   Which county had the highest percentage of its population sleeping under an insecticide treated mosquito net.
*   Which county has the least percentage of population sleeping under an insecticide treated mosquito net.


*   The county with highest percentage presenting symptoms .. is it the same county with the least mosquito net usage
*   The county with the least percentage presenting symptoms ..is it the same county with the highest bednet usage.

*   Top 5 Counties that are least in need of insecticide treated bednets.


*   Getting the county with the highest health spending per person.
*   Calculating the average health spending in Kenya


*   Counties that spend below the averge of health spending in Kenya.
*   Counties that spend above the averge of health spending in Kenya.

**Dataset Provided**

The dataset was sourced from the site **Kenya Open Data**

*   Health spending per capita by counties [[Link](https://www.opendata.go.ke/datasets/health-spending-per-capita-by-county/data)]






















### Importing the libraries.

In [0]:
# Import the dataset.
import numpy as np 
import pandas as pd


### Loading the Dataset.

In [0]:
# The dataframe for the malaria dataset.
df = pd.read_csv('Health_Spending_Per_Capita_By_County.csv')
df

County %_That_Slept_Under_a_Bed_Net  ... Geolocation  OBJECTID
0           Baringo                       24.20%  ...         NaN         0
1             Bomet                       26.30%  ...         NaN         1
2           Bungoma                       27.80%  ...         NaN         2
3             Busia                       40.70%  ...         NaN         3
4   Elgeyo/Marakwet                        7.50%  ...         NaN         4
5              Embu                       21.80%  ...         NaN         5
6           Garissa                       38.40%  ...         NaN         6
7          Homa Bay                       39.60%  ...         NaN         7
8            Isiolo                       42.60%  ...         NaN         8
9           Kajiado                       30.40%  ...         NaN         9
10         Kakamega                       26.70%  ...         NaN        10
11          Kericho                       33.80%  ...         NaN        11
12           Kiambu                       14.50%  ...         NaN        12
13           Kilifi                       38.50%  ...         NaN        13
14        Kirinyaga                          30%  ...         NaN        14
15            Kisii                       37.80%  ...         NaN        15
16           Kisumu                       62.40%  ...         NaN        16
17            Kitui                          13%  ...         NaN        17
18            Kwale                       32.10%  ...         NaN        18
19         Laikipia                       12.30%  ...         NaN        19
20             Lamu                       51.30%  ...         NaN        20
21         Machakos                       31.70%  ...         NaN        21
22          Makueni                       10.80%  ...         NaN        22
23          Mandera                        4.60%  ...         NaN        23
24         Marsabit                       11.10%  ...         NaN        24
25             Meru                          24%  ...         NaN        25
26           Migori                       40.10%  ...         NaN        26
27          Mombasa                       50.60%  ...         NaN        27
28         Murang'a                       11.40%  ...         NaN        28
29          Nairobi                          38%  ...         NaN        29
30           Nakuru                        9.80%  ...         NaN        30
31            Nandi                       18.90%  ...         NaN        31
32            Narok                       12.90%  ...         NaN        32
33          Nyamira                       40.90%  ...         NaN        33
34        Nyandarua                        1.70%  ...         NaN        34
35            Nyeri                        5.70%  ...         NaN        35
36          Samburu                        3.20%  ...         NaN        36
37            Siaya                       46.70%  ...         NaN        37
38     Taita Taveta                       30.80%  ...         NaN        38
39       Tana River                       45.10%  ...         NaN        39
40    Tharaka Nithi                       41.30%  ...         NaN        40
41      Trans Nzoia                          23%  ...         NaN        41
42          Turkana                           6%  ...         NaN        42
43      Uasin Gishu                       33.60%  ...         NaN        43
44           Vihiga                       28.50%  ...         NaN        44
45            Wajir                       18.30%  ...         NaN        45
46       West Pokot                       15.90%  ...         NaN        46

[47 rows x 6 columns]

In [0]:
# Viewing the first 5
df.head()

County %_That_Slept_Under_a_Bed_Net  ... Geolocation  OBJECTID
0          Baringo                       24.20%  ...         NaN         0
1            Bomet                       26.30%  ...         NaN         1
2          Bungoma                       27.80%  ...         NaN         2
3            Busia                       40.70%  ...         NaN         3
4  Elgeyo/Marakwet                        7.50%  ...         NaN         4

[5 rows x 6 columns]

### Data Preparation

In [0]:
# Checking the info of the data.
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 47 entries, 0 to 46
Data columns (total 6 columns):
 #   Column                         Non-Null Count  Dtype  
---  ------                         --------------  -----  
 0   County                         47 non-null     object 
 1   %_That_Slept_Under_a_Bed_Net   47 non-null     object 
 2   %_That_Had_a_Fever_or_Malaria  47 non-null     object 
 3   Heath_Spending_Per_Person_     47 non-null     int64  
 4   Geolocation                    0 non-null      float64
 5   OBJECTID                       47 non-null     int64  
dtypes: float64(1), int64(2), object(3)
memory usage: 2.3+ KB


In [0]:
# Checking if there are any null values.
df.isnull().sum()

County                            0
%_That_Slept_Under_a_Bed_Net      0
%_That_Had_a_Fever_or_Malaria     0
Heath_Spending_Per_Person_        0
Geolocation                      47
OBJECTID                          0
dtype: int64

In [0]:
# Getting the size of the dataframe
df.size

282

In [0]:
# Showing the size of the dataframe
df.shape

(47, 6)

In [0]:
# Using describe to show the statistics of the dataframe
df.describe()

Heath_Spending_Per_Person_  Geolocation   OBJECTID
count                   47.000000          0.0  47.000000
mean                    24.042553          NaN  23.000000
std                     11.500392          NaN  13.711309
min                      7.000000          NaN   0.000000
25%                     16.500000          NaN  11.500000
50%                     22.000000          NaN  23.000000
75%                     29.000000          NaN  34.500000
max                     50.000000          NaN  46.000000

### Data Cleaning

#### **Irrelevancy**

In [0]:
# IRRELEVANCY
# Irrelevant data are those that are not actually needed, and don’t fit under 
# the context of the problem we’re trying to solve.
df = df.drop(['Geolocation', 'OBJECTID'], axis = 1)


In [0]:
df.head()

County  ... Heath_Spending_Per_Person_
0          Baringo  ...                         29
1            Bomet  ...                         20
2          Bungoma  ...                         20
3            Busia  ...                         20
4  Elgeyo/Marakwet  ...                         48

[5 rows x 4 columns]

#### **Completeness**

In [0]:
# COMPLETENESS
# Checking/ Counting Missing Values 
# 

# Checking if there is any missing value in dataframe as a whole
# 
df.isnull()

# Checking if there is any missing value across each column
#
df.isnull().any()

# Checking how many missing values there are across each column
# 
df.isnull().sum()

# Or we can do a quick check to see if we have any missing values at all
#
print(df.isnull().values.any())

# We might also want to get a total count of missing values
# 
print(df.isnull().sum().sum())

False
0


#### **Consistency**

In [0]:
# CONSISTENCY
# Checking for duplicates that are repeated in my dataset. 
# These should be simply removed.
#
print(df.drop_duplicates())

             County  ... Heath_Spending_Per_Person_
0           Baringo  ...                         29
1             Bomet  ...                         20
2           Bungoma  ...                         20
3             Busia  ...                         20
4   Elgeyo/Marakwet  ...                         48
5              Embu  ...                         34
6           Garissa  ...                         37
7          Homa Bay  ...                         24
8            Isiolo  ...                         42
9           Kajiado  ...                         15
10         Kakamega  ...                         23
11          Kericho  ...                         11
12           Kiambu  ...                         20
13           Kilifi  ...                         26
14        Kirinyaga  ...                         18
15            Kisii  ...                         26
16           Kisumu  ...                         29
17            Kitui  ...                         46
18          

In [0]:
df.shape

(47, 4)

####**Uniformity**

In [0]:
# UNIFORMITY
# Standardization - Renaming column names
# Renaming the columns all to lowercase.
# 
cols = ['county','pct_that_slept_under_a_bed_net','pct_that_had_a_fever_or_malaria', 'health_spending_per_person']
df.columns = cols

In [0]:
# Confirming that the dataframe columns have been renamed appriopriately
df.columns
df.head()

county  ... health_spending_per_person
0          Baringo  ...                         29
1            Bomet  ...                         20
2          Bungoma  ...                         20
3            Busia  ...                         20
4  Elgeyo/Marakwet  ...                         48

[5 rows x 4 columns]

In [0]:
# Removing the % symbol from my dataset columns
#
df['pct_that_slept_under_a_bed_net'] = df['pct_that_slept_under_a_bed_net'].str.rstrip('%')
df['pct_that_had_a_fever_or_malaria'] = df['pct_that_had_a_fever_or_malaria'].str.rstrip('%')

In [0]:
# Checking if the changes are reflected on the dataframe.
df.head()

county  ... health_spending_per_person
0          Baringo  ...                         29
1            Bomet  ...                         20
2          Bungoma  ...                         20
3            Busia  ...                         20
4  Elgeyo/Marakwet  ...                         48

[5 rows x 4 columns]

In [0]:
# Final check for null before export.
#
df.isnull().sum()

county                             0
pct_that_slept_under_a_bed_net     0
pct_that_had_a_fever_or_malaria    0
health_spending_per_person         0
dtype: int64

####**Exporting the dataset**

In [0]:
# Exporting the cleaned dataset
df.to_csv('health_counties.csv')

### Analysis

#### Loading the SQL extension

In [0]:
##
## Loading the sql
#
%load_ext sql

%sql sqlite://

'Connected: @None'

#### Loading the CSV dataset

In [0]:
##
## 
# Loading our table from the respective CSV files 
with open('health_counties.csv','r') as f:
    counties = pd.read_csv(f, index_col=0, encoding='utf-8')
%sql DROP TABLE if EXISTS counties;
%sql PERSIST counties;
%sql SELECT * FROM counties LIMIT 5;

 * sqlite://
Done.
 * sqlite://
 * sqlite://
Done.


[(0, 'Baringo', 24.2, 14.9, 29),
 (1, 'Bomet', 26.3, 39.5, 20),
 (2, 'Bungoma', 27.8, 37.6, 20),
 (3, 'Busia', 40.7, 42.8, 20),
 (4, 'Elgeyo/Marakwet', 7.5, 9.4, 48)]

In [0]:
## Which county had the highest % of population sleeping under a bednet
##
%%sql 
SELECT county, MAX(pct_that_slept_under_a_bed_net) AS high_pct_slept_under_net 
FROM counties;

 * sqlite://
Done.


[('Kisumu', 62.4)]



> The county that had the highest percentage of population who are sleeping under a net is **Kisumu** with **62.4%**





In [0]:
## County with the least % sleeping under a bed net
##
%%sql
SELECT county, MIN(pct_that_slept_under_a_bed_net) AS low_pct_slept_under_net
FROM counties;

 * sqlite://
Done.


[('Nyandarua', 1.7)]



> The county that had the lowest percentage of population who are sleeping under a net is **Nyandarua** with **1.7%**



In [0]:
## County with highest % presenting with symptopms .. is it the same county with the least bednet usage
##
%%sql
SELECT county, MAX(pct_that_had_a_fever_or_malaria) 
FROM counties;

 * sqlite://
Done.


[('Lamu', 63.3)]



> The county with the highest percentage of malaria **Lamu** isn't the same county with the least percentage of net usage **Nyandarua**



In [0]:
##  County with the least % presenting with symptoms ..is it the same county with the highest bednet usage
##
%%sql
SELECT county, MIN(pct_that_had_a_fever_or_malaria) 
FROM counties;

 * sqlite://
Done.


[('Nakuru', 0.0)]



> The county with the least percentage of malaria **Nakuru** isn't the same county with the highest percentage of net usage **Kisumu**



In [0]:
## Average percentage of those who slept under a bed net.
##
%%sql
SELECT AVG(pct_that_slept_under_a_bed_net) AS avg_sleep_under_net
FROM counties;

 * sqlite://
Done.


[(26.72978723404255,)]

In [0]:
## Average percentage of those who have malaria.
##
%%sql
SELECT AVG(pct_that_had_a_fever_or_malaria) AS avg_malaria
FROM counties;

 * sqlite://
Done.


[(40.1808510638298,)]

In [0]:
## Top 5 Counties that are really in need of insecticide treated bednets.
##
%%sql
SELECT county, pct_that_slept_under_a_bed_net, pct_that_had_a_fever_or_malaria 
FROM counties
WHERE pct_that_had_a_fever_or_malaria > (SELECT AVG(pct_that_had_a_fever_or_malaria) FROM counties)
ORDER BY pct_that_had_a_fever_or_malaria DESC;

 * sqlite://
Done.


[('Lamu', 51.3, 63.3),
 ('Tana River', 45.1, 63.3),
 ('Migori', 40.1, 63.0),
 ('West Pokot', 15.9, 60.4),
 ('Homa Bay', 39.6, 58.3),
 ('Narok', 12.9, 55.8),
 ('Turkana', 6.0, 55.4),
 ('Wajir', 18.3, 54.8),
 ('Siaya', 46.7, 54.4),
 ('Kericho', 33.8, 51.6),
 ('Makueni', 10.8, 51.1),
 ('Nandi', 18.9, 50.9),
 ('Kitui', 13.0, 50.5),
 ('Kisii', 37.8, 49.0),
 ('Taita Taveta', 30.8, 48.8),
 ('Mombasa', 50.6, 48.0),
 ('Garissa', 38.4, 46.6),
 ('Kisumu', 62.4, 45.6),
 ('Kilifi', 38.5, 45.4),
 ('Marsabit', 11.1, 44.0),
 ('Uasin Gishu', 33.6, 43.4),
 ('Kirinyaga', 30.0, 43.2),
 ('Busia', 40.7, 42.8),
 ('Meru', 24.0, 42.6),
 ('Machakos', 31.7, 40.9)]

In [0]:
## Top 5 Counties that are least in need of insecticide treated bednets.
##
%%sql
SELECT county, pct_that_slept_under_a_bed_net, pct_that_had_a_fever_or_malaria
FROM counties
WHERE pct_that_had_a_fever_or_malaria < (SELECT AVG(pct_that_had_a_fever_or_malaria) FROM counties)
ORDER BY pct_that_had_a_fever_or_malaria ASC;

 * sqlite://
Done.


[('Nakuru', 9.8, 0.0),
 ('Elgeyo/Marakwet', 7.5, 9.4),
 ('Kajiado', 30.4, 12.3),
 ('Nyeri', 5.7, 13.4),
 ('Nyandarua', 1.7, 14.4),
 ('Baringo', 24.2, 14.9),
 ('Laikipia', 12.3, 14.9),
 ('Kiambu', 14.5, 22.8),
 ('Vihiga', 28.5, 33.4),
 ('Isiolo', 42.6, 33.7),
 ('Tharaka Nithi', 41.3, 33.8),
 ('Trans Nzoia', 23.0, 34.8),
 ('Mandera', 4.6, 35.2),
 ("Murang'a", 11.4, 36.2),
 ('Samburu', 3.2, 37.1),
 ('Bungoma', 27.8, 37.6),
 ('Kwale', 32.1, 37.7),
 ('Nairobi', 38.0, 37.7),
 ('Kakamega', 26.7, 37.9),
 ('Nyamira', 40.9, 38.9),
 ('Bomet', 26.3, 39.5),
 ('Embu', 21.8, 39.8)]

In [0]:
## Getting the county with the highest health spending per person
## 
%%sql
SELECT county,pct_that_slept_under_a_bed_net, pct_that_had_a_fever_or_malaria, MAX(health_spending_per_person) AS high_health_spending
FROM counties;

 * sqlite://
Done.


[('Lamu', 51.3, 63.3, 50)]



> The county with the highest health spending per person is **Lamu** at **50%**



In [0]:
## Getting the county with the least health spending per person.
##
#
%%sql
SELECT county,pct_that_slept_under_a_bed_net, pct_that_had_a_fever_or_malaria, MIN(health_spending_per_person) AS least_health_spending
FROM counties;

 * sqlite://
Done.


[('Nairobi', 38.0, 37.7, 7)]



> The county with the lowest health spending per person is **Nairobi** at **7%**



In [0]:
## Calculating the average health spending in Kenya
## 
#
%%sql
SELECT AVG(health_spending_per_person) AS avg_spending
FROM counties;

 * sqlite://
Done.


[(24.04255319148936,)]



> The average health spending in Kenya is at **24%**



In [0]:

## Counties that spend below the averge of health spending in Kenya.
## 
#
%%sql
SELECT county, pct_that_slept_under_a_bed_net, pct_that_had_a_fever_or_malaria, health_spending_per_person 
FROM counties 
WHERE health_spending_per_person < (SELECT AVG(health_spending_per_person) FROM counties)
ORDER BY health_spending_per_person ASC;

 * sqlite://
Done.


[('Nairobi', 38.0, 37.7, 7),
 ('Narok', 12.9, 55.8, 7),
 ('Trans Nzoia', 23.0, 34.8, 8),
 ('Vihiga', 28.5, 33.4, 10),
 ('Kericho', 33.8, 51.6, 11),
 ('Mombasa', 50.6, 48.0, 11),
 ('Nakuru', 9.8, 0.0, 11),
 ('Uasin Gishu', 33.6, 43.4, 13),
 ('Turkana', 6.0, 55.4, 14),
 ('Kajiado', 30.4, 12.3, 15),
 ('Makueni', 10.8, 51.1, 15),
 ('Machakos', 31.7, 40.9, 16),
 ('Nyandarua', 1.7, 14.4, 17),
 ('Kirinyaga', 30.0, 43.2, 18),
 ('Tharaka Nithi', 41.3, 33.8, 18),
 ('Tana River', 45.1, 63.3, 19),
 ('Bomet', 26.3, 39.5, 20),
 ('Bungoma', 27.8, 37.6, 20),
 ('Busia', 40.7, 42.8, 20),
 ('Kiambu', 14.5, 22.8, 20),
 ('Mandera', 4.6, 35.2, 20),
 ('Wajir', 18.3, 54.8, 21),
 ('Meru', 24.0, 42.6, 22),
 ("Murang'a", 11.4, 36.2, 22),
 ('Nyamira', 40.9, 38.9, 22),
 ('Siaya', 46.7, 54.4, 22),
 ('Kakamega', 26.7, 37.9, 23),
 ('Migori', 40.1, 63.0, 23),
 ('Nandi', 18.9, 50.9, 23),
 ('Homa Bay', 39.6, 58.3, 24),
 ('West Pokot', 15.9, 60.4, 24)]

In [0]:
## Counties that spend above the average of health spending in Kenya.
## 
%%sql
SELECT county, pct_that_slept_under_a_bed_net, pct_that_had_a_fever_or_malaria, health_spending_per_person
FROM counties 
WHERE health_spending_per_person > (SELECT AVG(health_spending_per_person) FROM counties)
ORDER BY health_spending_per_person DESC;

 * sqlite://
Done.


[('Lamu', 51.3, 63.3, 50),
 ('Elgeyo/Marakwet', 7.5, 9.4, 48),
 ('Taita Taveta', 30.8, 48.8, 48),
 ('Kitui', 13.0, 50.5, 46),
 ('Marsabit', 11.1, 44.0, 46),
 ('Isiolo', 42.6, 33.7, 42),
 ('Nyeri', 5.7, 13.4, 39),
 ('Garissa', 38.4, 46.6, 37),
 ('Embu', 21.8, 39.8, 34),
 ('Samburu', 3.2, 37.1, 34),
 ('Kwale', 32.1, 37.7, 31),
 ('Baringo', 24.2, 14.9, 29),
 ('Kisumu', 62.4, 45.6, 29),
 ('Laikipia', 12.3, 14.9, 29),
 ('Kilifi', 38.5, 45.4, 26),
 ('Kisii', 37.8, 49.0, 26)]

In [0]:
%%sql
SELECT * 
FROM counties

 * sqlite://
Done.


[(0, 'Baringo', 24.2, 14.9, 29),
 (1, 'Bomet', 26.3, 39.5, 20),
 (2, 'Bungoma', 27.8, 37.6, 20),
 (3, 'Busia', 40.7, 42.8, 20),
 (4, 'Elgeyo/Marakwet', 7.5, 9.4, 48),
 (5, 'Embu', 21.8, 39.8, 34),
 (6, 'Garissa', 38.4, 46.6, 37),
 (7, 'Homa Bay', 39.6, 58.3, 24),
 (8, 'Isiolo', 42.6, 33.7, 42),
 (9, 'Kajiado', 30.4, 12.3, 15),
 (10, 'Kakamega', 26.7, 37.9, 23),
 (11, 'Kericho', 33.8, 51.6, 11),
 (12, 'Kiambu', 14.5, 22.8, 20),
 (13, 'Kilifi', 38.5, 45.4, 26),
 (14, 'Kirinyaga', 30.0, 43.2, 18),
 (15, 'Kisii', 37.8, 49.0, 26),
 (16, 'Kisumu', 62.4, 45.6, 29),
 (17, 'Kitui', 13.0, 50.5, 46),
 (18, 'Kwale', 32.1, 37.7, 31),
 (19, 'Laikipia', 12.3, 14.9, 29),
 (20, 'Lamu', 51.3, 63.3, 50),
 (21, 'Machakos', 31.7, 40.9, 16),
 (22, 'Makueni', 10.8, 51.1, 15),
 (23, 'Mandera', 4.6, 35.2, 20),
 (24, 'Marsabit', 11.1, 44.0, 46),
 (25, 'Meru', 24.0, 42.6, 22),
 (26, 'Migori', 40.1, 63.0, 23),
 (27, 'Mombasa', 50.6, 48.0, 11),
 (28, "Murang'a", 11.4, 36.2, 22),
 (29, 'Nairobi', 38.0, 37.7, 7),
 (30, 'Nakuru', 9.8, 0.0, 11),
 (31, 'Nandi', 18.9, 50.9, 23),
 (32, 'Narok', 12.9, 55.8, 7),
 (33, 'Nyamira', 40.9, 38.9, 22),
 (34, 'Nyandarua', 1.7, 14.4, 17),
 (35, 'Nyeri', 5.7, 13.4, 39),
 (36, 'Samburu', 3.2, 37.1, 34),
 (37, 'Siaya', 46.7, 54.4, 22),
 (38, 'Taita Taveta', 30.8, 48.8, 48),
 (39, 'Tana River', 45.1, 63.3, 19),
 (40, 'Tharaka Nithi', 41.3, 33.8, 18),
 (41, 'Trans Nzoia', 23.0, 34.8, 8),
 (42, 'Turkana', 6.0, 55.4, 14),
 (43, 'Uasin Gishu', 33.6, 43.4, 13),
 (44, 'Vihiga', 28.5, 33.4, 10),
 (45, 'Wajir', 18.3, 54.8, 21),
 (46, 'West Pokot', 15.9, 60.4, 24)]

ERROR! Session/line number was not unique in database. History logging moved to new session 59
